Source/RSP/Interpreter Ops.c

```diff
@@ -584,7 +584,7 @@ void RSP_Vector_VMUDN (void) {
 	for (el = 0; el < 8; el ++ ) {
 		del = EleSpec[RSPOpC.rs].B[el];
 
-		temp.UW = (DWORD)RSP_Vect[RSPOpC.rd].UHW[el] * (DWORD)(long)(RSP_Vect[RSPOpC.rt].HW[del]);
+		temp.UW = (uint32_t)((uint32_t)RSP_Vect[RSPOpC.rd].UHW[el]) * (uint32_t)(( int32_t)RSP_Vect[RSPOpC.rt].HW[del]);

```

Comment:
in (uint32_t)((uint32_t)RSP_Vect[RSPOpC.rd].UHW[el]) isn't the second (uint32_t) redundant